src/Tools/ExternalAccess/Extensions/External/DocumentLinePosition.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text.Json.Serialization;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Extensions;
+
+/// <summary>
+/// Immutable representation of a line number and position within a document.
+/// </summary>
+public readonly struct DocumentLinePosition
```

Comment:
doc (maybe in remarks) that this isn't part of Roslyn, but is actually intended to be a gladstone concept?